Додавання базових кнопок

MaksStud opened the pull request:
Додано запуск бота
Додано функцію відповіді на команду start
Додано основні кнопки

MaksStud committed:
Додано запуск бота
Додано функцію відповіді на команду start
Додано основеі кнопки

andrsj commented on create_bot.py:
```diff
@@ -0,0 +1,5 @@
+from aiogram import Bot, Dispatcher
+
+TOKEN = '6726585823:AAE7FCNk_8DlM1Zktj3m07nn2JaoDz27KI4'
```
Не пуш токен в репозиторій! Цей токен потрібно брати з файлу або змінних оточень
(Якщо не знаєш як - напиши мені, направлю)

MaksStud committed:
Додано ехо функцію

andrsj commented on main.py:
```diff
@@ -0,0 +1,20 @@
+
+if __name__ == "__main__":
+    user.register(dp)
+    logging.basicConfig(level=logging.INFO, stream=sys.stdout)
```
На перший час нам треба `DEBUG` рівень, `INFO` слугуватиме нам в проді
Цей рівень я б також виніс у файл чи змінні оточення як з токеном

andrsj commented on user/user.py:
```diff
@@ -0,0 +1,22 @@
+from aiogram.filters.command import Command
+from aiogram.filters.state import StatesGroup, State
+from aiogram.fsm.context import FSMContext
+from . import message_text as m
```
Не використовуй назви в одну букву - це знижує читабельність

andrsj commented on user/user.py:
```diff
@@ -0,0 +1,22 @@
+
+
+async def command_start(message: Message) -> None:
+    await message.answer(m.hello_message, reply_markup=kb.hello_kb)
```
Не вистачає розбиття на рівні
Я розумію що це лише відповідь з статичним повідомленням, але я хочу взяти код за основу і його розширювати

Будь твоя ласка напиши `service` (або `usecase`) який повертає цей текст, а в цьому методі просто викличи цей сервіс

Якщо не зрозумів - напиши мені

andrsj commented on user/user.py:
```diff
@@ -0,0 +1,22 @@
+
+
+async def eho(message: Message) -> None:
+    await message.answer(message.text)
```
Аналогічно з попереднім методом

andrsj reviewed (changes requested):
Не вистачає `.gitignore` файлу
Не вистачає `requirements.txt` файлу
Не вистачає `virtual env` налаштувань (якщо не зрозумів про що я - напиши мені)
Опиши в `README.md` файлі як запускати код, бо на даний момент запуск досить простий, але це лише зараз така ситуація

andrsj commented:
Загалом модуль `user` збиває з толку, адже код написаний саме для бота, а не користувача
Пропоную перейменувати в `bot`

andrsj commented on create_bot.py:
```diff
@@ -0,0 +1,5 @@
+from aiogram import Bot, Dispatcher
+
+TOKEN = '6726585823:AAE7FCNk_8DlM1Zktj3m07nn2JaoDz27KI4'
+dp = Dispatcher()
```
Я б переніс створення всіх залежностей в main file, щоб всі класи утворювались в одному місці

andrsj replied:
Додай type для змінної

andrsj commented on create_bot.py:
```diff
@@ -0,0 +1,5 @@
+
+TOKEN = '6726585823:AAE7FCNk_8DlM1Zktj3m07nn2JaoDz27KI4'
+dp = Dispatcher()
+bot = Bot(TOKEN, parse_mode="html")
```
Тут також type додати

andrsj commented on user/user.py:
```diff
@@ -0,0 +1,22 @@
+    await message.answer(m.hello_message, reply_markup=kb.hello_kb)
+
+
+async def eho(message: Message) -> None:
```
`echo` - typo

MaksStud committed:
Додано опис як запустити проект

Create README.md

andrsj commented on README.md:
```diff
@@ -0,0 +1,2 @@
+## Запуск коду
+Для запуску проекту запускаємо файл main.py
```
Окей, поки ти не додав `virt env` та `req.txt` - то це підійде, але в проді такого опису в нас не буде

MaksStud committed:
Виправнено помилки

Додано:
.gitignore
requirements.txt
клас для повідомлень

Видалено:
папку user

Змінено:
logging.INFO на logging.DEBUG

MaksStud committed:
Merge remote-tracking branch 'origin/basic-buttons' into basic-buttons

MaksStud committed:
Виправлено назву функції
eho на echo

MaksStud committed:
Додано тип для змінної

MaksStud committed:
Видалив .idea 

Delete .idea directory

andrsj commented on main.py:
```diff
@@ -0,0 +1,20 @@
+import asyncio
+import logging
+import sys
+
```
Тут варто оголосити всі залежності, що потрібні для роботи (виклик методів чи конструкторів - не важливо)

MaksStud replied:
Не зовсім розумію

andrsj replied:
Створити всі залежності в головному файлі:
- бот
- диспетчер
- сервіс
- логер

andrsj commented on user.py:
```diff
@@ -0,0 +1,20 @@
+
+
+async def command_start(message: Message) -> None:
+    user_text = User_message_text()
```
Трішки не правильно утворені залежності
Ти на кожну обробку створюєш новий об'єкт, на даний момент це не є проблема...
...АЛЕ, коли в нас з'являться більш тяжчі залежності (база та брокери повідомлень) - це буде критично, адже треба буде створювати нові залежності кожного разу

MaksStud replied:
Чомусь коли я поза функцією створював екземпляр класу то в мене виникала помилка і я не знаю як її виправити

andrsj replied:
@MaksStud не достатньо контексту - яка помилка, при якому коді. Я не можу дати відповідь без цього котнексту

MaksStud replied:
Я зміг її виправити

MaksStud committed:
Винесено user_text за функцію

MaksStud committed:
Merge remote-tracking branch 'origin/basic-buttons' into basic-buttons

MaksStud committed:
додано клас App

andrsj commented on message_text.py:
```diff
@@ -0,0 +1,10 @@
+from aiogram.types import Message
```
Для чого тобі потрібен цей клас?

MaksStud replied:
Вже виправив

MaksStud committed:
Видалив лишній рядок коду

andrsj commented on main.py:
```diff
@@ -0,0 +1,9 @@
+if __name__ == "__main__":
+    app = App(TOKEN)
+    logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
+    asyncio.run(app.run())
```
Я думаю `asyncio.run` можна перекинути в сам метод, а тут лише викликати метод

andrsj commented on user.py:
```diff
@@ -0,0 +1,20 @@
+user_text = User_message_text()
+
+
+async def command_start(message: Message) -> None:
+    await message.answer(user_text.hello_user_message(), reply_markup=kb.hello_kb)
+
+
+async def echo(message: Message) -> None:
+    await message.answer(user_text.echo_user_message(message.text))
```
Їх б об'єднати в клас `Handlers` з залежностями:

```python
class Handlers:
    def __init__(self, user_text, keyboards):
        self.user_text = user_text
        self.keyboards = keyboards

    async def command_start(self, message: Message) -> None:
        await message.answer(self.user_text.hello_user_message(), reply_markup=self.keyboards.hello_kb)

    async def echo(self, message: Message) -> None:
        await message.answer(self.user_text.echo_user_message(message.text))
```

Щось типу цього (не забудь про typing)

andrsj commented on user.py:
```diff
@@ -0,0 +1,20 @@
+    await message.answer(user_text.echo_user_message(message.text))
+
+
+def register(dp: Dispatcher):
+    dp.message.register(command_start, Command('start'))
+    dp.message.register(echo)
```
Цей код повинен бути на старті додатку

andrsj commented on create_bot.py:
```diff
@@ -0,0 +1,26 @@
+import logging
+import sys
+
+TOKEN: str = '6726585823:AAE7FCNk_8DlM1Zktj3m07nn2JaoDz27KI4'
```
Environment variable!

andrsj commented on keyboards.py:
```diff
@@ -0,0 +1,7 @@
+                [InlineKeyboardButton(text='Підклюити копіювання', callback_data='Підклюити копіювання')],
+                [InlineKeyboardButton(text='Продовжити підписку', callback_data='Продовжити підписку')],]
+
+hello_kb = InlineKeyboardMarkup(inline_keyboard=hello_buttons)
```
Давай ми обійдемось без глобальних змінних
Варто переписати це на метод, який генерує ці кнопки
